ampd/src/main.rs

```diff
@@ -119,16 +136,19 @@ fn parse_config(
         .map_err(Report::from)
 }
 
-fn check_state_path(path: &Path) -> error_stack::Result<PathBuf, Error> {
-    expand_home_dir(path)
-        .then(canonicalize)
-        .change_context(Error::StateLocation(path.to_string_lossy().into_owned()))
-}
-
-fn expand_home_dir(path: &Path) -> PathBuf {
+fn expand_home_dir(path: impl AsRef<Path>) -> PathBuf {
+    let path = path.as_ref();
     let Ok(home_subfolder) = path.strip_prefix("~") else{
         return path.to_path_buf()
     };
 
     dirs::home_dir().map_or(path.to_path_buf(), |home| home.join(home_subfolder))
 }
+
+#[derive(Error, Debug)]
+enum Error {
+    #[error("failed to load config")]
+    LoadConfig,
+    #[error("catastrophic failure")]
```

Comment:
```
#[error("fatal failure")]
```
catastrophic failure sounds a little extreme. Would make me worry that the cpu melted down or something like that.